mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1169,43 +1169,138 @@ def _log_eval_table(self):
             )
         else:
             data = self.dataset.features_data.assign(outputs=self.y_pred)
+        data = data.assign(**self.metrics_dict)
         mlflow.log_table(data, artifact_file=f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}")
 
-    def _evaluate_question_answering(self):
-        self._log_eval_table()
-        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
-        self.artifacts[name] = JsonEvaluationArtifact(
-            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+    def _calculate_perplexity(self, predictions):
+        try:
+            import evaluate
+
+            perplexity = evaluate.load("perplexity", module_type="metric")
+        except Exception as e:
+            _logger.warning(
+                f"Failed to load 'perplexity' metric (error: {e!r}), skipping metric logging."
+            )
+            return
+
+        results = perplexity.compute(predictions=predictions, model_id="gpt2")
+        self.metrics.update({"mean_perplexity": results["mean_perplexity"]})
+        self.metrics_dict.update({"perplexity": results["perplexities"]})
+
+    def _calculate_toxicity(self, predictions):
+        try:
+            from transformers import (
+                AutoModelForSequenceClassification,
+                AutoTokenizer,
+                TextClassificationPipeline,
+            )
+
+            model_path = "martin-ha/toxic-comment-model"
+            tokenizer = AutoTokenizer.from_pretrained(model_path)
+            model = AutoModelForSequenceClassification.from_pretrained(model_path)
+
+            pipeline = TextClassificationPipeline(model=model, tokenizer=tokenizer)
+        except Exception as e:
+            _logger.warning(
+                f"Failed to load 'toxicity' metric (error: {e!r}), skipping metric logging."
+            )
+            return
+
+        results = pipeline(list(predictions))
+        percent_toxic = {
+            "percent_toxic": sum([1 if result["label"] == "toxic" else 0 for result in results])
+            / len(results)
+        }
+        self.metrics.update(percent_toxic)
+        self.metrics_dict.update({"toxicity": results})
+
+    def _calculate_reading_level(self, predictions):
+        try:
+            import textstat
+        except Exception as e:
+            _logger.warning(
+                f"Failed to load reading level metrics (error: {e!r}), skipping metric logging."
+            )
+            return
```

Comment:
```suggestion
        try:
            import textstat
        except ImportError:
            _logger.warning(
                f"Failed to load reading level metrics (error: {e!r}), skipping metric logging."
            )
            return
```

Can we use `ImportError` and update the message?